sdk/lib/_internal/js_runtime/lib/js_mirrors.dart

```diff
@@ -612,11 +620,27 @@ TypeMirror reflectClassByName(Symbol symbol, String mangledName) {
   if (typeArgIndex != -1) {
     TypeMirror originalDeclaration =
         reflectClassByMangledName(mangledName.substring(0, typeArgIndex))
-        .originalDeclaration;
+            .originalDeclaration;
     if (originalDeclaration is JsTypedefMirror) {
       throw new UnimplementedError();
     }
-    mirror = new JsTypeBoundClassMirror(originalDeclaration,
+    if (originalDeclaration.typeVariables.isEmpty) {
+      throw new ArgumentError.value(originalDeclaration.reflectedType);
+    }
+    List<String> typeArguments = mangledName
+        .substring(typeArgIndex + 1, mangledName.length - 1)
+        .split(', ')
+        .where((_) => _.isNotEmpty)
```

Comment:
I agree it's awkward. Unfortunately it's the way `split` works on empty strings (results in `[""]`).

I moved `isEmpty` check on one level up the chain, since this would only be possible with a class name like `Foo<>`, which is not entirely valid. So preventing such behavior seems like a better solution to me.
5fdc5b9.